services/worker-manager/src/providers/azure.js

```diff
@@ -707,47 +713,60 @@ class AzureProvider extends Provider {
     this.errors[worker.workerPoolId] = this.errors[worker.workerPoolId] || [];
     let state = worker.state || states.REQUESTED;
     try {
-      const { provisioningState } = await this.fetchVmInfo(worker);
       // lets us get power states for the VM
       const instanceView = await this._enqueue('get', () => this.computeClient.virtualMachines.instanceView(
         worker.providerData.resourceGroupName,
         worker.providerData.vm.name,
       ));
       const powerStates = instanceView.statuses.map(i => i.code);
       monitor.debug({
-        message: 'fetched instance states',
+        message: 'fetched instance view',
         powerStates,
-        provisioningState,
       });
-      if (successProvisioningStates.has(provisioningState) &&
-          // fairly lame check, succeeds if we've ever been starting/running
-          _.some(powerStates, v => successPowerStates.has(v))
-      ) {
-        this.seen[worker.workerPoolId] += worker.capacity || 1;
-
-        if (worker.providerData.terminateAfter && worker.providerData.terminateAfter < Date.now()) {
-          state = await this.removeWorker({ worker, reason: 'terminateAfter time exceeded' });
-        }
-      } else if (failProvisioningStates.has(provisioningState) ||
-                // if the VM has ever been in a failing power state
-                _.some(powerStates, v => failPowerStates.has(v))
-      ) {
-        state = await this.removeWorker({
-          worker,
-          reason: `failed state; provisioningState=${provisioningState}, powerStates=${powerStates.join(', ')}`,
-        });
-      } else {
-        const { workerPoolId } = worker;
-        const workerPool = await WorkerPool.get(this.db, workerPoolId);
-        if (workerPool) {
-          await this.reportError({
-            workerPool,
-            kind: 'creation-error',
-            title: 'Encountered unknown VM provisioningState or powerStates',
-            description: `Unknown provisioningState ${provisioningState} or powerStates: ${powerStates.join(', ')}`,
-          });
+
+      // count this worker as having been seen for later logging
+      this.seen[worker.workerPoolId] += worker.capacity || 1;
```

Comment:
Note that this count _was_ used for provisioning decisions, but that's no longer the case -- the provisioning loop now counts workers directly.  This is only used in the `logSeen` call in this file.  So that it now counts VMs while they are starting up is not going to cause any issues.